Main.java

```diff
@@ -4,26 +4,42 @@
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class Main {
 
-    final static String FILE_TYPE_TXT = ".txt";
-    final static String FILE_TYPE_JSON = ".json";
-    final static String REMOVE_SPECIAL_CHAR = "\\p{Punct}";
-    final static String SINGLE_SPACE = "";
+    static final String FILE_TYPE_TXT = ".txt";
+    static final String FILE_TYPE_JSON = ".json";
+    static final String REMOVE_SPECIAL_CHAR = "\\p{Punct}";
+    static final String SINGLE_SPACE = "";
+    static final String EXCLUDE_SPECIAL_CHAR = "[a-zA-Z0-9\"\"]";
 
     public static void main(String[] args) {
 
-        if (args.length != 2) {
+        if (args.length < 2) {
             System.out.println("File path and Search word cannot be empty");
             return;
         }
+        else if(args.length > 2){
+            System.out.println("There can only be two arguments");
+            return;
+        }
         String textFilePath = args[0];
         String searchKeyword = args[1];
+        Matcher matcher = Pattern.compile(REMOVE_SPECIAL_CHAR).matcher(searchKeyword);

```

Comment:
Why is punctuation check made on user input? We can just return the searchword not found right? User can pass anything as input.

Reply:
Even if the user inputs something like "@vibecheck" and there exists a word "@vibecheck" in the file, our code will only return "Word does not exist" since we are removing all the special characters from the file contents. Hence the punctuation check.

Reply:
Got it, that makes sense, but the same applies if the file contains a word like say 'hi-tech' in the file and user search for hi-tech, but will not find it since hi-tech will be read as hi tech in the program right? Or hyphen words are not separated?

Reply:
Got it. I will alter the code to not separate hyphen words.